### ISSUE TITLE
Update these models

### ISSUE THREAD

**Issue by 22i** [maikerumine/mobs_mc#134]
@Wuzzy2  i tried to update more mobs to switch from overlays to more textures but was not very succesfull could you update these mobs below

- [x] pig

model https://github.com/22i/amc/blob/master/models/mobs_mc_pig.b3d

uses two textures now - pig and saddle
https://github.com/22i/amc/blob/master/textures/mobs_mc_pig.png
https://github.com/22i/amc/blob/master/textures/mobs_mc_pig_saddle.png

lua https://github.com/22i/amc/blob/master/init.lua#L1167

- [ ] sheep

model https://github.com/22i/amc/blob/master/models/mobs_mc_sheep.b3d

uses two textures now
https://github.com/22i/amc/blob/master/textures/mobs_mc_sheep.png
https://github.com/22i/amc/blob/master/textures/mobs_mc_sheep_fur.png

lua https://github.com/22i/amc/blob/master/init.lua#L1479

- [ ] snowman

model https://github.com/22i/amc/blob/master/models/mobs_mc_snowman.b3d

now uses many textures one for body every other for its pumpkin head
https://github.com/22i/amc/blob/master/textures/mobs_mc_snowman.png
https://github.com/22i/amc/blob/master/textures/amc_pumpkin_face_off.png
https://github.com/22i/amc/blob/master/textures/amc_pumpkin_side.png
https://github.com/22i/amc/blob/master/textures/amc_pumpkin_top.png

lua https://github.com/22i/amc/blob/master/init.lua#L1630

- [ ] vex

model https://github.com/22i/amc/blob/master/models/mobs_mc_vex.b3d

uses 2 textures one for body other for weapon
https://github.com/22i/amc/blob/master/textures/mobs_mc_vex.png
https://github.com/22i/amc/blob/master/textures/mobs_mc_iron_sword.png

lua https://github.com/22i/amc/blob/master/init.lua#L1712

- [ ] horse

model https://github.com/22i/amc/blob/master/models/mobs_mc_horse.b3d

texture still uses overlay for armour and now horse base texture includes saddle and chest
https://github.com/22i/amc/blob/master/textures/mobs_mc_horse_zombie.png
https://github.com/22i/amc/blob/master/textures/mobs_mc_horse_white.png
https://github.com/22i/amc/blob/master/textures/mobs_mc_horse_skeleton.png
https://github.com/22i/amc/blob/master/textures/mobs_mc_horse_gray.png
https://github.com/22i/amc/blob/master/textures/mobs_mc_horse_darkbrown.png
https://github.com/22i/amc/blob/master/textures/mobs_mc_horse_creamy.png
https://github.com/22i/amc/blob/master/textures/mobs_mc_horse_chestnut.png
https://github.com/22i/amc/blob/master/textures/mobs_mc_horse_brown.png
https://github.com/22i/amc/blob/master/textures/mobs_mc_horse_black.png

lua https://github.com/22i/amc/blob/master/init.lua#L2254

- [ ] zombie pigman

model https://github.com/22i/amc/blob/master/models/mobs_mc_zombie_pigman.b3d

uses 2 textures now one for body other for weapon
https://github.com/22i/amc/blob/master/textures/mobs_mc_zombie_pigman.png
https://github.com/22i/amc/blob/master/textures/amc_gold_sword.png

lua https://github.com/22i/amc/blob/master/init.lua#L2170

- [ ] skeleton

model https://github.com/22i/amc/blob/master/models/mobs_mc_skeleton.b3d

uses texture for body and texture for a weapon
https://github.com/22i/amc/blob/master/textures/mobs_mc_skeleton.png
https://github.com/22i/amc/blob/master/textures/mobs_mc_bow_standby.png

lua https://github.com/22i/amc/blob/master/init.lua#L1961

animation
`shoot_start = 70, shoot_end = 90, die_start = 160, die_end = 170, die_speed = 15`

- [ ] stray

model https://github.com/22i/amc/blob/master/models/mobs_mc_skeleton_stray.b3d

uses 3 textures now one for body other for stray clothes last one for weapon
https://github.com/22i/amc/blob/master/textures/mobs_mc_stray.png
https://github.com/22i/amc/blob/master/textures/mobs_mc_stray_overlay.png
https://github.com/22i/amc/blob/master/textures/mobs_mc_bow_standby.png

lua https://github.com/22i/amc/blob/master/init.lua#L2048

animation
`shoot_start = 70, shoot_end = 90, die_start = 160, die_end = 170, die_speed = 15`

- [ ] wither skeleton

model https://github.com/22i/amc/blob/master/models/mobs_mc_wither_skeleton.b3d

uses 2 textures now one for body other for weapon
https://github.com/22i/amc/blob/master/textures/mobs_mc_wither_skeleton.png
https://github.com/22i/amc/blob/master/textures/mobs_mc_stone_sword.png

lua https://github.com/22i/amc/blob/master/init.lua#L2005

animation
`punch_start = 110, punch_end = 130, die_start = 160, die_end = 170, die_speed = 15`

**Comment by Wuzzy2** [comment]
Yeah, I think Iook at this later, it's a lot of work, and I like to work on mapgen now.

But I of course totally agree this is needed.

**Comment by 22i** [comment]
take your time no need to rush

**Comment by Wuzzy2** [comment]
Pig is done.

I'm not sure what to do with the weapons, since they are now identical to the actual weapon inventory images. I'm not too happy with duplicating these.

I'm not sure if I can use the updated horse stuff in this form, especially since the armor layer is missing.
Sheep will probably need the most work.

So: Looks like you work on changing the textures in a way that they are compatible with MC texture packs. This is of course greatly appreciated.

**Comment by 22i** [comment]
> I'm not sure what to do with the weapons, since they are now identical to the actual weapon inventory images. I'm not too happy with duplicating these.

What do you mean? do you now know how to make mobs hold inventory items without adding 3d items in blender?

> I'm not sure if I can use the updated horse stuff in this form, especially since the armor layer is missing.

For armour i can make an extra armour layer or you could use overlay armour on horse base texture
`{
            "blank.png", --chest
            "mobs_mc_horse_brown.png^mobs_mc_horse_armor_gold.png", --horse base texture
            "blank.png", --saddle
        },`

> Pig is done.

sweet!

**Comment by Wuzzy2** [comment]
About the weapons: That's not what I meant! I was thinking loudly of whether I include the redundant weapon texture files.

**Comment by 22i** [comment]
oh that you could just use the ones you have no need to have 2 of em

redundant textures could be used during changing weapon textures during that time there should be a redundant weapon texture till the time it is recreated in 3d and the model is updated

**Comment by Wuzzy2** [comment]
Snowman is done in MCL2 (but not in the standalone `mobs_mc` mod), it uses now the pumpkin textures in the `mcl_farming` mod.

I tried to figure out multiple times how to get your new sheep working, but I just didn't succeed. Can you please help me out here? :D

Note I have abandoned `mobs_mc` as a standalone mod. It just doubled the work for me and frankly, this mod never really fit well into Minetest Game anyway. Instead, I will prefer to work on the mods directly in MCL2.

As for the remaining mobs (vex, horse, zombie pigman, skeleton, stray, wither skeleton), I will probably not adopt them as you submitted them, mainly because they introduce non-standard overlays. In this form I do not like to adopt them, sorry. Ideally, mob texturing should be done in a way that textures from a Minecraft texture pack would work.

**Comment by 22i** [comment]
yep sheep are complicated will try to look into them

can you check again? if you download https://github.com/22i/amc/archive/master.zip and check under the textures you will not see lots of overlays like in the past versions

what do you mean with non-standard overlays?

i think these are compatible with mc texture packs

i dont think they are using any overlays check https://github.com/22i/amc/blob/master/init.lua
-search for ^

except enderman uses them for its eyes and horses for horse armour

**Comment by Wuzzy2** [comment]
I have looked into this again and I'm happy to say that I managed to put your re-done models into MCL2!
It actually wasn't as problematic as I thought.

Note this standalone is unaffected.

**Comment by 22i** [comment]
congratz!
